src/mixins/object_interactivity.mixin.js

```diff
@@ -43,21 +43,44 @@
           continue;
         }
 
-        lines = this._getImageLines(this.oCoords[i].corner);
+        lines = fabric.util.object.clone(this._getImageLines(this.oCoords[i].corner), true);
+
+        var t = 'topline',
+            b = 'bottomline',
+            l = 'leftline',
+            r = 'rightline',
+            size = this.cornerRegionSize;
+
+        lines[t].d.x += size;
+        lines[t].o.x -= size;
+        lines[t].d.y += size;
+        lines[t].o.y += size;
+        lines[b].d.x += size;
+        lines[b].o.x -= size;
+        lines[b].d.y -= size;
+        lines[b].o.y -= size;
+        lines[l].d.x -= size;
+        lines[l].o.x -= size;
+        lines[l].d.y -= size;
+        lines[l].o.y += size;
+        lines[r].d.x += size;
+        lines[r].o.x += size;
+        lines[r].d.y += size;
+        lines[r].o.y -= size;
```

Comment:
ok so i see some red flags here.
We are cloning the data in a performance loop and adding the new size here.
We can do that when we calculate those points first time.